gdk/build_system/Zip.py

```diff
@@ -41,19 +43,11 @@ def build(self, **kwargs):
             logging.debug("Copying over component files to the '{}' folder.".format(artifacts_zip_build.name))
             root_directory_path = utils.get_current_directory()
 
-            def ignore_patterns_in_root(path, names):
-                if str(path) == str(root_directory_path):
-                    return {
-                        name
-                        for pattern in self.get_ignored_file_patterns(project_config)
-                        for name in fnmatch.filter(names, pattern)
-                    }
-                return set()
-
-            shutil.copytree(
+            custom_copytree.copytree(
```

Comment:
why not keep using shutil, but with a custom filter that looks at the full path, rather than only the filename?

Reply:
Yea at first I thought this wouldn't work but realized it would work after all. See latest commit.